test/e2e/persistentips-primary_test.go

```diff
@@ -116,6 +116,8 @@ var _ = Describe("Persistent IPs on Primary UDN interface", func() {
 					WithTimeout(5 * time.Minute).
 					Should(testenv.ContainConditionVMIReady())
 
+				Expect(testenv.Client.Get(context.Background(), client.ObjectKeyFromObject(vmi), vmi)).To(Succeed())
```

Comment:
please consider making it part of the Eventually block
(also on the other places)
this what we are usually on kubevirt fwiw

Reply:
I don't think we need eventually, because this call is right after we wait for the VMI to be ready

Reply:
you already fetch the latest vmi as part of `ThisVMI` (implicit Eventually happens for what you need)
so you can just use that value externally

func ThisVMI(vmi *kubevirtv1.VirtualMachineInstance) func() (*kubevirtv1.VirtualMachineInstance, error) {
	return func() (p *kubevirtv1.VirtualMachineInstance, err error) {
		p = &kubevirtv1.VirtualMachineInstance{}
		err = Client.Get(context.Background(), client.ObjectKeyFromObject(vmi), p)
		if apierrors.IsNotFound(err) {
			return nil, nil
		}
		return
	}
}

Reply:
true, but this change has bigger scope than what I want to achieve in this PR. 
But we should definitely should do it on a follow up PR (change ThisVMI + remove all the "fetch" that is used before the assertion)

Reply:
honestly i dont understand why to not add 
`vmi = testenv.ThisVMI(vmi)` in the first line of the Eventually and thats all

Reply:
I intend to do so, but in a larger scope, in a follow up PR